state/pool.go

```diff
@@ -355,7 +367,12 @@ func (p *StatePool) Remove(modelUUID string) (bool, error) {
 		// ignore unknown model uuids.
 		return false, nil
 	}
-	item.remove = true
+	if !item.remove {
+		item.remove = true
+		if item.removing != nil {
+			close(item.removing)
+		}
+	}
```

Comment:
It would be potentially safer to use a `sync.Once` here, but again it might be overkill.

Also, when can `item.removing` equal nil? We never set it to nil and a PoolItem is always created, so I think we can omit the `if item.removing != nil {` check.

Reply:
It is only nil when it is the system state. But I see the `if modelUUID == p.systemState.ModelUUID()` at the start of the method, so I could probably drop it, but strictly, it is possible to be nil.

Reply:
Also on the sync once suggestion, this is behind the `p.mu` mutex lock, so effectively the same.